### DISCUSSION TITLE
Overrides in Spectral 6

### DISCUSSION THREAD

**Discussion by krousseau** [stoplightio/spectral#1942] [Q&A] [answered]
Hi all,

I am trying to migrate from v5 to v6 and am getting hung up on overrides. Any format I try for `files` is throwing the following error:
```
Ruleset must have some source assigned. If you use Spectral programmatically make sure to pass the source to Ruleset
```
Everything works fine (besides the list of errors I receive) when I don't include overrides.

Here's the example of what I'm doing 
ruleset.js
```
overrides: [{
    "files": ["../../generated/openapi3/api.json#/paths/foo/delete/requestBody/content/application~1vnd.segment.v1alpha+json/example/rules/0"],
    "rules": {
      "delete-request-format": "off"
    }
  }]
```
Note, I've also tried many different paths (including `*`, `**/*.json`, `api.json`, `**/api.json`)

The following worked in v5 in `ruleset.json`:
```
"except": {
    "../../generated/openapi3/api.json#/paths/foo/delete/requestBody/content/application~1vnd.segment.v1alpha+json/example/rules/0": [
      "delete-request-format"
    ],
```

My validate function does the following:
```
  const document = new Document(openApiJson, JsonParser, '/some/absolute/path/generated/openapi3/api.json')
  const spectral = new Spectral()
  await setSpectralRuleset(spectral, path.join(__dirname, 'ruleset.js'))
  const results = await spectral.run(document)
```

**Comment by krousseau** [comment, marked as answer]
After a bit more exploration and digging into the code I found that I needed to do the following:
```
  const m = require(rulesetPath)
  const rulesetFML = new Ruleset(m, { source: openApiSpecPath })
```

If this is actually required for overrides, then it would be great if it can be added to the docs.

**Reply by dantuch** [reply]
Hello @krousseau can you share a bit more details? How did you manage to make this work?

**Comment by dantuch** [comment]
Hello, I don't think that accepted answer is complete and so - helpful to others. 

Can spectral team try to find a moment and clarify wether this issue still exists and if so - come up with a better explanation how to solve it? 

Here OP has his own context in head, used methods and variables without showing thiers declaration etc - it's hard to follow up.
